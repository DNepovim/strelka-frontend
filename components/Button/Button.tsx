import styled from "@emotion/styled"

import Wrapper from "../../assets/vectors/potatoes/rectangle_2.svg"
import { theme } from "../../styles/theme"

export const Button = styled.button`
  font-family: ${theme.fonts.accent};
  font-weight: bold;
  font-size: ${theme.size.medium}rem;
  mask-image: url("${Wrapper.src}");
  mask-size: 100% 100%;
  mask-position: center;
  mask-repeat: no-repeat;
  background-color: ${theme.color.darkAccent};
  color: ${theme.color.lightest};
  border: none;
`
