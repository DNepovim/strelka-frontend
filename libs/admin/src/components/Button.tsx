import styled from "@emotion/styled"
import { buttonHover, theme } from "../theme"
import { css } from "@emotion/react"
import React, { ButtonHTMLAttributes } from "react"
import { RouteLink } from "./RouteLink"

export interface ButtonProps {
  danger?: boolean
  circled?: boolean
}

export const Button: React.FC<
  ButtonHTMLAttributes<HTMLButtonElement> & ButtonProps
> = (props) => <StyledButton type="button" {...props} />

const StyledButton = styled.button`
  display: flex;
  align-items: start;
  cursor: pointer;
  font-size: 1rem;
  padding: 0.4em 1em;
  border: ${theme.styles.border};
  background-color: white;
  transition: background-color 200ms;
  text-decoration: none;
  color: ${({ danger }: ButtonProps) =>
    danger ? theme.colors.danger : theme.colors.text};
  ${({ circled }: ButtonProps) =>
    circled
      ? css`
          border-radius: 1em;
        `
      : ""}

  svg {
    margin: 0 0.4em 0 -0.2em;
  }
`

export const ButtonLink = StyledButton.withComponent(RouteLink)
