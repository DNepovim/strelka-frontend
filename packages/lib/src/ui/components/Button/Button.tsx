/** @jsxImportSource @emotion/react */
import React from "react"
import { css } from "@emotion/react"
import { theme } from "../../styles/theme"

export interface ButtonProps {
  link: string
  targetBlank?: boolean
  isSmall?: boolean
}

export const Button: React.FC<ButtonProps> = ({
  children,
  link,
  targetBlank,
  isSmall,
}) => (
  <a
    href={link}
    {...(targetBlank
      ? {
          target: "_blank",
          rel: "noreferrer noopener",
        }
      : {})}
    css={css`
      position: relative;
      display: inline-block;
      font-family: themix;
      font-size: ${isSmall ? 0.8 : 1.4}rem;
      color: ${theme.color.background};
      padding: 0.6em 1em;
      border-radius: 4px;
      text-decoration: none;
      background-color: ${theme.color.brand};
      overflow: hidden;

      &:before {
        content: "";
        position: absolute;
        top: 0;
        left: 0;
        width: 0;
        height: 100%;
        background-color: ${theme.color.second};
        transition: width 700ms cubic-bezier(0.1, 0.15, 0, 0.97);
      }

      span {
        position: relative;
        z-index: 10;
      }

      &:hover {
        text-decoration: none;

        &:before {
          width: 100%;
        }
      }
      @media (min-width: 400px) {
        font-size: ${isSmall ? 1 : 1.4}rem;
      }
    `}
  >
    <span>{children}</span>
  </a>
)

export default Button
