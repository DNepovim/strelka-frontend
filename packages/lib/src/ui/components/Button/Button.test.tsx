import renderer from "react-test-renderer"
import { Button } from "./Button"

describe("Button component", () => {
  it("should renders correctly", () => {
    const tree = renderer
      .create(<Button link="http://www.somewhere.com">Link somewhere</Button>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should renders correctly with target blank", () => {
    const tree = renderer
      .create(
        <Button link="http://www.somewhere.com" targetBlank>
          Link somewhere
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should renders correctly small button", () => {
    const tree = renderer
      .create(
        <Button link="http://www.somewhere.com" isSmall>
          Link somewhere
        </Button>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
