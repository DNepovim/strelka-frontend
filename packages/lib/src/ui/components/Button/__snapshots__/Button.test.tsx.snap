// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should renders correctly 1`] = `
.emotion-0 {
  position: relative;
  display: inline-block;
  font-family: themix;
  font-size: 1.4rem;
  color: #ffffff;
  padding: 0.6em 1em;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #ccc15a;
  overflow: hidden;
}

.emotion-0:before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 0;
  height: 100%;
  background-color: #aac15a;
  -webkit-transition: width 700ms cubic-bezier(0.1, 0.15, 0, 0.97);
  transition: width 700ms cubic-bezier(0.1, 0.15, 0, 0.97);
}

.emotion-0 span {
  position: relative;
  z-index: 10;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover:before {
  width: 100%;
}

@media (min-width: 400px) {
  .emotion-0 {
    font-size: 1.4rem;
  }
}

<a
  className="emotion-0"
  href="http://www.somewhere.com"
>
  <span>
    Link somewhere
  </span>
</a>
`;

exports[`Button component should renders correctly small button 1`] = `
.emotion-0 {
  position: relative;
  display: inline-block;
  font-family: themix;
  font-size: 0.8rem;
  color: #ffffff;
  padding: 0.6em 1em;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #ccc15a;
  overflow: hidden;
}

.emotion-0:before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 0;
  height: 100%;
  background-color: #aac15a;
  -webkit-transition: width 700ms cubic-bezier(0.1, 0.15, 0, 0.97);
  transition: width 700ms cubic-bezier(0.1, 0.15, 0, 0.97);
}

.emotion-0 span {
  position: relative;
  z-index: 10;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover:before {
  width: 100%;
}

@media (min-width: 400px) {
  .emotion-0 {
    font-size: 1rem;
  }
}

<a
  className="emotion-0"
  href="http://www.somewhere.com"
>
  <span>
    Link somewhere
  </span>
</a>
`;

exports[`Button component should renders correctly with target blank 1`] = `
.emotion-0 {
  position: relative;
  display: inline-block;
  font-family: themix;
  font-size: 1.4rem;
  color: #ffffff;
  padding: 0.6em 1em;
  border-radius: 4px;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #ccc15a;
  overflow: hidden;
}

.emotion-0:before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 0;
  height: 100%;
  background-color: #aac15a;
  -webkit-transition: width 700ms cubic-bezier(0.1, 0.15, 0, 0.97);
  transition: width 700ms cubic-bezier(0.1, 0.15, 0, 0.97);
}

.emotion-0 span {
  position: relative;
  z-index: 10;
}

.emotion-0:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-0:hover:before {
  width: 100%;
}

@media (min-width: 400px) {
  .emotion-0 {
    font-size: 1.4rem;
  }
}

<a
  className="emotion-0"
  href="http://www.somewhere.com"
  rel="noreferrer noopener"
  target="_blank"
>
  <span>
    Link somewhere
  </span>
</a>
`;
